### PR TITLE
Extract Qwen3-ASR mel at true audio length instead of padding to 3000 mel frames

### PR DESCRIPTION
**Pull request by 0xjeffro**
## Motivation

Qwen3-ASR reuses Whisper's feature extractor, which pads every audio clip to 30s (3000 mel frames) before computing the mel spectrogram. SeedTTS clips are only ~4–7s, so most of that CPU mel/FFT work runs on silence.

Profiling the c=32 path shows per-request preprocessing (request_build, ~28ms) is a big chunk of wall-clock and runs serially before the GPU (https://github.com/sgl-project/sglang-omni/issues/831#issuecomment-4756457401). But Qwen3-ASR's audio encoder is variable-length and already drops the padded frames via feature_attention_mask, so the 30s padding is pure waste.

## Modifications

Extract the mel at the clip's true length instead of padding to 30s

## Related Issues

#831

#### End-to-end benchmark

| metric | before | after | Δ |
|---|---|---|---|
| throughput (samples/s) | 22.43 | 27.88 | **+24.3%** |
| wall mean (s) | 48.52 | 39.04 | −19.5% |
| latency mean (s) | 1.416 | 1.144 | −19.2% |
| latency p95 (s) | 1.925 | 1.546 | −19.7% |
| rtf mean | 0.3064 | 0.2474 | −19.3% |
| corpus WER (mean of 3) | 0.01297 | 0.01310 | unchanged (noise) |
| skipped | 0 | 0 | — |

WER per-repeat:
before: 0.0129/0.0128/0.0132; after: 0.0138/0.0127/0.0128.

#### Stage breakdown (mechanism)

| interval | before avg | after avg | Δ |
|---|---|---|---|
| `scheduler_request_build_start→end` | 27.9 ms | 18.5 ms | **−34%** |
| `scheduler_request_build_start→end` p95 | 37.6 ms | 27.0 ms | −28% |
| `stage_input_received→stage_complete` (end-to-end, incl. queue) | 1561 ms | 1285 ms | −18% |

---

## Checklist

- [x] Format your code according with pre-commit.
- [ ] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

CI runs on self-hosted GPU runners and requires a maintainer to add the
`run-ci` label. Once labeled, every subsequent push re-triggers CI as
long as the label remains. Use `/tag-and-rerun-ci higgs` or
`/tag-and-rerun-ci moss` to select a TTS CI model. Draft PRs are skipped even
if labeled.

### CODE REVIEW

File: sglang_omni/models/qwen3_asr/request_builders.py [hunk 1]
```diff
@@ -171,22 +171,31 @@ def request_builder(payload: StagePayload) -> Qwen3ASRRequestData:
...
         feature_attention_mask = getattr(extracted, "attention_mask", None)
         if feature_attention_mask is None:
             # WhisperFeatureExtractor normally returns one; fall back to all-valid.
             feature_attention_mask = torch.ones(
                 (features.shape[0], features.shape[-1]), dtype=torch.long
             )
-        # Keep the full padded mel; the model's get_audio_feature uses the mask
-        # to select valid frames. Its no-mask branch transposes wrong, so the
-        # mask path must be taken.
+        # get_audio_feature uses the mask to select valid frames; its no-mask
```
Comment (zhaochenyang20) [inline sglang_omni/models/qwen3_asr/request_builders.py]: add `note (Jeffro Qu))`

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]

**Comment by zhaochenyang20** [comment]
compared with this:

image

https://github.com/sgl-project/sglang-omni/actions/runs/27839788427/job/82401275576

image


Good PR for Qwen3 ASR.

**Review by zhaochenyang20 — APPROVED** [review]
